tests/test_io.py

```diff
@@ -29,6 +29,92 @@ def test_read_yaml() -> None:
     TestCase().assertDictEqual(sample_config, CONFIG)
 
 
+def test_find_images() -> None:
+    """Test finding images"""
+    found_images = find_images(base_dir="tests/", file_ext=".spm")
+
+    assert isinstance(found_images, list)
+    assert len(found_images) == 1
+    assert isinstance(found_images[0], Path)
+    assert "minicircle.spm" in str(found_images[0])
+
+
+@pytest.mark.parametrize(
+    "base_dir, image_path, output_dir, expected",
+    [
+        # Absolute path, nested under base_dir, with file suffix
+        (
+            Path("/some/random/path"),
+            Path("/some/random/path/images/test.spm"),
+            Path("output/here"),
+            Path("output/here/images/test/"),
+        ),
+        # Absolute path, nested under base_dir, with file suffix and multiple periods
+        (
+            Path("/some/random/path"),
+            Path("/some/random/path/images/te.st.spm"),
+            Path("output/here"),
+            Path("output/here/images/te.st/"),
+        ),
+        # Absolute path, nested under base_dir, with file suffix
+        (
+            Path("/some/random/path"),
+            Path("/some/random/path/images/today/test.spm"),
+            Path("output/here"),
+            Path("output/here/images/today/test/"),
+        ),
+        # Relative path, nested under base_dir, with file_suffix
+        (
+            Path("/some/random/path"),
+            Path("images/test.spm"),
+            Path("output/here"),
+            Path("output/here/images/test"),
+        ),
+        # Relative path, nested (two deep) under base_dir, with file_suffix
+        (
+            Path("/some/random/path"),
+            Path("images/today/test.spm"),
+            Path("output/here"),
+            Path("output/here/images/today/test"),
+        ),
+        # Relative path, nested under base_dir, no file suffix
+        (Path("/some/random/path"), Path("images/"), Path("output/here"), Path("output/here/images/")),
+        # Absolute path, nested under base_dir, output not nested under base_dir, with file_suffix
```

Comment:
Here it says `output not nested under base_dir`, what does this mean in contrast to the other tests?

Reply:
Ahah I think I get it. Including `/` or not switches between forcing the output to be just what the output dir is set as (with the `folder_name/image_name` directory added), or appending the input file structure on top of the output structure.

Reply:
Correct :smiley: